fluss-common/src/main/java/com/alibaba/fluss/utils/PartitionUtils.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2024 Alibaba Group Holding Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.fluss.utils;
+
+import com.alibaba.fluss.config.AutoPartitionTimeUnit;
+import com.alibaba.fluss.exception.PartitionSpecInvalidException;
+import com.alibaba.fluss.types.DataTypeRoot;
+
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static com.alibaba.fluss.utils.Preconditions.checkArgument;
+
+/** Utils for partition. */
+public class PartitionUtils {
+    public static final String PARTITION_SPEC_SEPARATOR = "$";
+
+    public static final List<DataTypeRoot> STATIC_PARTITION_KEY_SUPPORTED_TYPES =
+            Arrays.asList(
+                    DataTypeRoot.STRING,
+                    DataTypeRoot.INTEGER,
+                    DataTypeRoot.BOOLEAN,
+                    DataTypeRoot.DATE,
+                    DataTypeRoot.TIME_WITHOUT_TIME_ZONE);
```

Comment:
Let's support other data types in other PRs. Currently, let's only support STRING type. The problem here is the partition value -> string conversion. There is no a standard how to convert a DATE or TIME into string, the Java `Data/Time#toString()` is not a common standard and usually has problems in partitions (e.g., contains special characters `:`).

Reply:
Ok, I have created a issue: https://github.com/alibaba/fluss/issues/489 to trace this, also I added a `todo` comment in code.